{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "xHkTPkr9PetF"
      },
      "source": [
        "# Mounting Google Drive"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "ebWJoqkqOS6a"
      },
      "outputs": [],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "PAO6HKpHpJyK"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "\n",
        "PROJECT_DIR=\"/content/drive/MyDrive/BioinformaticsProject\""
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7yxCQUjKPEja"
      },
      "source": [
        "# Installing Fasterq-Dump"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "j9o5SVj5KUq5"
      },
      "outputs": [],
      "source": [
        "# remove files\n",
        "!rm -rd sratoolkit*/\n",
        "!rm sratoolkit/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "FdLTdoroInbh"
      },
      "outputs": [],
      "source": [
        "# Install fasterq-dump through the sratoolkit\n",
        "!wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
        "!tar -xzf sratoolkit.current-ubuntu64.tar.gz\n",
        "!mv sratoolkit.*-ubuntu64 sratoolkit\n",
        "!export PATH=$PATH:/content/sratoolkit/bin"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3D1zs1xaK3fe"
      },
      "outputs": [],
      "source": [
        "# Call this once in order to be able to call fasterq-dump beyond the first cell\n",
        "import os\n",
        "os.environ[\"PATH\"] += \":/content/sratoolkit/bin\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "AcVmgQoUMS7Q"
      },
      "outputs": [],
      "source": [
        "# Check that fasterq is installed\n",
        "!fasterq-dump --version"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QAkd-MBBPQIB"
      },
      "source": [
        "# Installing Micromamba"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Gg-JScfFNcTQ"
      },
      "source": [
        "Following cells install micromamba and create the enviorenment we'll use"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "wEcyCLnCnGO_"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "# Install micromamba into /usr/local/bin\n",
        "cd /usr/local\n",
        "\n",
        "curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest \\\n",
        "  -o micromamba.tar.bz2\n",
        "\n",
        "tar -xvjf micromamba.tar.bz2 bin/micromamba\n",
        "\n",
        "# Move it into /usr/local/bin so it's on PATH\n",
        "#mv bin/micromamba /usr/local/bin/\n",
        "#rmdir bin"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "T36FwQuiIDkS"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "# Enable micromamba's shell integration\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "\n",
        "# Create the environment (run once per notebook/session)\n",
        "micromamba create -y -n beeasm python=3.10"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "OUWlugP0RFQz"
      },
      "source": [
        "#"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "f_gho981gZVs"
      },
      "outputs": [],
      "source": [
        "#@title CARPENTER BEE DUMP\n",
        "!fasterq-dump SRR24955310 \\\n",
        "  -O /content/CarpenterBee_fastq \\\n",
        "  --temp /content/fasterq_tmp \\\n",
        "  -e 6 -p\n",
        "#!ls -ld \"${PROJECT_DIR}/Reads/CarpenterBee\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "i_L0h3Lz2eDF"
      },
      "outputs": [],
      "source": [
        "#@title HONEY BEE DUMP\n",
        "!fasterq-dump SRR36076821 -e 6 -p"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wc -l \"SRR24955310_1.fastq\"\n",
        "!wc -l \"SRR24955310_2.fastq\""
      ],
      "metadata": {
        "id": "niGL2zL0HAXB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "!cp /content/SRR24955310_1.fastq /content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee/new_reads\n",
        "!cp /content/SRR24955310_2.fastq /content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee/new_reads\n"
      ],
      "metadata": {
        "id": "Irk-vcmYqEFR"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR36076821_1.fastq /content/drive/MyDrive/BioinformaticsProject/Reads/Honeybee/new_reads\n",
        "!cp /content/SRR36076821_2.fastq /content/drive/MyDrive/BioinformaticsProject/Reads/Honeybee/new_reads"
      ],
      "metadata": {
        "id": "rW2HPFiZ1jzY"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "DAH-HING1-AK"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "W5GUZTmgvWCd"
      },
      "outputs": [],
      "source": [
        "#@title CLEAR WORKING DIRECTORY\n",
        "import shutil\n",
        "import os\n",
        "\n",
        "# items to keep\n",
        "keep = {\"drive\", \".config\", \".ipynb_checkpoints\", \"sratoolkit\"}\n",
        "\n",
        "for name in os.listdir(\"/content\"):\n",
        "    if name in keep:\n",
        "        continue\n",
        "\n",
        "    path = os.path.join(\"/content\", name)\n",
        "\n",
        "    if os.path.isdir(path):\n",
        "        shutil.rmtree(path)\n",
        "    else:\n",
        "        os.remove(path)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9gNUXIiPNCC2"
      },
      "source": [
        "Following cell installs the spades assembler\n",
        "\n",
        "> Add blockquote\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "O6VTSzqQM3u1"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "# Enable micromamba in this shell\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# Install SPAdes from bioconda\n",
        "micromamba install -y -c bioconda -c conda-forge spades\n",
        "\n",
        "# Quick sanity check\n",
        "spades.py --version\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4GDWT6CUQM8d"
      },
      "source": [
        "Run spades on our bacteria"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "7GYeBOVDNtj0"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "set -e\n",
        "\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# One Illumina dataset (single-end from R1)\n",
        "READS=\"/content/drive/MyDrive/BioinformaticsProject/Reads/Bacteria/SRR3924617_1.fastq\"\n",
        "OUTDIR=\"${PROJECT_DIR}/Reads/CarpenterBee/result_spades\"\n",
        "THREADS=8    # adjust if needed\n",
        "MEM_GB=12    # approximate RAM limit\n",
        "\n",
        "\n",
        "# extract a partial read\n",
        "SUBREADS=\"/content/subsamble_5000.fastq\"\n",
        "NREADS=5000\n",
        "\n",
        "# Extract the first N reads\n",
        "head -n $((NREADS * 4)) \"$READS\" > \"$SUBREADS\"\n",
        "\n",
        "mkdir -p \"$OUTDIR\"\n",
        "\n",
        "spades.py \\\n",
        "  -s \"$READS\" \\\n",
        "  -o \"$OUTDIR\" \\\n",
        "  -t \"$THREADS\" \\\n",
        "  -m \"$MEM_GB\" \\\n",
        "  --careful\n",
        "\n",
        "echo \"SPAdes finished. Assembly:\"\n",
        "ls -lh \"$OUTDIR/contigs.fasta\"\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "B2qf9tDI4PrB"
      },
      "source": [
        "and now run spades on carpenter bee"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "rm -rf /content/result_spades_honeybee*\n",
        "mkdir /content/result_spades_honeybee"
      ],
      "metadata": {
        "id": "t5LhG_eRio_w"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Check reads with fast qc"
      ],
      "metadata": {
        "id": "q3wQGDSV5pPc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update\n",
        "!apt-get install -y fastqc"
      ],
      "metadata": {
        "id": "btu5fFIY5zm_",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc /content/SRR36076821_1.fastq /content/SRR36076821_2.fastq -o /content/qc_reports/"
      ],
      "metadata": {
        "id": "5Hhy2JIs5mN9",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "KD_PluTzB5YK"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Trimmomatic"
      ],
      "metadata": {
        "id": "3YI_9H3mCEjg"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "This tool is a part of our preprocessing stage of trimming the whole genome."
      ],
      "metadata": {
        "id": "yW4d2RLRJBlx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Installation\n",
        "\n",
        "%%bash\n",
        "apt-get update\n",
        "apt-get install -y trimmomatic default-jre\n"
      ],
      "metadata": {
        "id": "CITdaMmRCHFP",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "ls -lh /usr/share/trimmomatic"
      ],
      "metadata": {
        "id": "40TPCBM4DwSC"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1) Trim whole genome - Honeybee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "R1=\"/content/SRR36076821_1.fastq\"\n",
        "R2=\"/content/SRR36076821_2.fastq\"\n",
        "\n",
        "OUT1P=\"/content/SRR36076821_1.trimmed.fq\"\n",
        "OUT1U=\"/content/SRR36076821_1.unpaired.fq\"\n",
        "OUT2P=\"/content/SRR36076821_2.trimmed.fq\"\n",
        "OUT2U=\"/content/SRR36076821_2.unpaired.fq\"\n",
        "\n",
        "TRIMJAR=\"/usr/share/java/trimmomatic.jar\"\n",
        "ADAPTERS=\"/usr/share/trimmomatic/TruSeq3-PE.fa\"\n",
        "\n",
        "java -jar \"$TRIMJAR\" PE -threads 8 \\\n",
        "  \"$R1\" \"$R2\" \\\n",
        "  \"$OUT1P\" \"$OUT1U\" \\\n",
        "  \"$OUT2P\" \"$OUT2U\" \\\n",
        "  ILLUMINACLIP:${ADAPTERS}:2:30:10 \\\n",
        "  SLIDINGWINDOW:4:20 \\\n",
        "  MINLEN:50\n",
        "\n",
        "echo \"Trimmed reads:\"\n",
        "ls -lh \"$OUT1P\" \"$OUT2P\"\n"
      ],
      "metadata": {
        "id": "oz1Lxpt-COmf",
        "cellView": "form",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 2) Generate subreads - Honeybee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "micromamba install -y -c bioconda -c conda-forge seqtk\n",
        "\n",
        "R1=/content/SRR36076821_1.trimmed.fq\n",
        "R2=/content/SRR36076821_2.trimmed.fq\n",
        "\n",
        "NREADS=500000\n",
        "\n",
        "SUB1=\"/content/SRR36076821_1.sub${NREADS}.fq\"\n",
        "SUB2=\"/content/SRR36076821_2.sub${NREADS}.fq\"\n",
        "\n",
        "echo \"Creating subsampled reads (${NREADS} pairs)...\"\n",
        "\n",
        "echo \"Using seqtk for random subsampling.\"\n",
        "seqtk sample -s100 \"$R1\" \"$NREADS\" > \"$SUB1\"\n",
        "seqtk sample -s100 \"$R2\" \"$NREADS\" > \"$SUB2\"\n",
        "\n",
        "echo \"Subsampled files:\"\n",
        "ls -lh \"$SUB1\" \"$SUB2\""
      ],
      "metadata": {
        "cellView": "form",
        "id": "JP2Lju3aV4l_"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1) Trim whole genome - Carpenter Bee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "R1=\"/content/SRR24955310_1.fastq\"\n",
        "R2=\"/content/SRR24955310_2.fastq\"\n",
        "\n",
        "OUT1P=\"/content/SRR24955310_1.trimmed.fq\"\n",
        "OUT1U=\"/content/SRR24955310_1.unpaired.fq\"\n",
        "OUT2P=\"/content/SRR24955310_2.trimmed.fq\"\n",
        "OUT2U=\"/content/SRR24955310_2.unpaired.fq\"\n",
        "\n",
        "TRIMJAR=\"/usr/share/java/trimmomatic.jar\"\n",
        "ADAPTERS=\"/usr/share/trimmomatic/TruSeq3-PE.fa\"\n",
        "\n",
        "java -jar \"$TRIMJAR\" PE -threads 8 \\\n",
        "  \"$R1\" \"$R2\" \\\n",
        "  \"$OUT1P\" \"$OUT1U\" \\\n",
        "  \"$OUT2P\" \"$OUT2U\" \\\n",
        "  ILLUMINACLIP:${ADAPTERS}:2:30:10 \\\n",
        "  SLIDINGWINDOW:4:20 \\\n",
        "  MINLEN:50\n",
        "\n",
        "echo \"Trimmed reads:\"\n",
        "ls -lh \"$OUT1P\" \"$OUT2P\"\n"
      ],
      "metadata": {
        "collapsed": true,
        "cellView": "form",
        "id": "u5jtYri6Ifzj"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 2) Generate subreads - Carpenter Bee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "micromamba install -y -c bioconda -c conda-forge seqtk\n",
        "\n",
        "R1=/content/SRR24955310_1.trimmed.fq\n",
        "R2=/content/SRR24955310_2.trimmed.fq\n",
        "\n",
        "NREADS=500000\n",
        "\n",
        "SUB1=\"/content/SRR24955310_1.sub${NREADS}.fq\"\n",
        "SUB2=\"/content/SRR24955310_2.sub${NREADS}.fq\"\n",
        "\n",
        "echo \"Creating subsampled reads (${NREADS} pairs)...\"\n",
        "\n",
        "echo \"Using seqtk for random subsampling.\"\n",
        "seqtk sample -s100 \"$R1\" \"$NREADS\" > \"$SUB1\"\n",
        "seqtk sample -s100 \"$R2\" \"$NREADS\" > \"$SUB2\"\n",
        "\n",
        "echo \"Subsampled files:\"\n",
        "ls -lh \"$SUB1\" \"$SUB2\""
      ],
      "metadata": {
        "cellView": "form",
        "collapsed": true,
        "id": "PEygprag3_f3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Spades Assembler - Honeybee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# One Illumina dataset (single-end from R1)\n",
        "\n",
        "#OUTDIR=\"${PROJECT_DIR}/Reads/Honeybee/result_spades\"\n",
        "THREADS=8    # adjust if needed\n",
        "MEM_GB=12    # approximate RAM limit\n",
        "\n",
        "\n",
        "# extract a partial read\n",
        "#SUBREADS=\"/content/subsamble_5000.fastq\"\n",
        "#NREADS=5000\n",
        "\n",
        "# Extract the first N reads\n",
        "#head -n $((NREADS * 4)) \"$READS\" > \"$SUBREADS\"\n",
        "\n",
        "#mkdir -p \"$OUTDIR\"\n",
        "\n",
        "spades.py \\\n",
        "  -1 \"\" \\\n",
        "  -2 \"\" \\\n",
        "  -o \"/content/result_spades_honeybee\" \\\n",
        "  -t \"$THREADS\" \\\n",
        "  -m \"$MEM_GB\" \\\n",
        "  --careful\n",
        "\n",
        "echo \"SPAdes finished. Assembly:\"\n",
        "ls -lh \"$OUTDIR/contigs.fasta\"\n"
      ],
      "metadata": {
        "id": "eeiiQRJHK-G4",
        "cellView": "form",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "p3A95zENK93B"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "D9aKdErpsrq5"
      },
      "source": [
        "# Megahit Installation\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "82QqFlo9sxUa"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# Install MEGAHIT\n",
        "micromamba install -y -c bioconda -c conda-forge megahit\n",
        "\n",
        "# Sanity check\n",
        "megahit --version\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "which megahit\n"
      ],
      "metadata": {
        "id": "aMsno9HeUfi3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CXX24mk9s9Lf"
      },
      "source": [
        "and this cell runs it on bacteria"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Y5m7FEOrtVLA",
        "cellView": "form",
        "collapsed": true
      },
      "outputs": [],
      "source": [
        "#@title Megahit - bacteria\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# Full dataset (single-end)\n",
        "READS=\"/content/drive/MyDrive/BioinformaticsProject/Reads/Bacteria/SRR3924617_1.fastq\"\n",
        "\n",
        "# Subset: first 5000 reads -> 4 * 5000 = 20000 lines\n",
        "SUB=\"/content/SRR3924617_1.first5000.fastq\"\n",
        "OUTDIR=\"/content/megahit_bacteria_5k\"\n",
        "THREADS=4\n",
        "\n",
        "echo \"Creating subset with first 5000 reads...\"\n",
        "sed -n '1,20000p' \"$READS\" > \"$SUB\"\n",
        "ls -lh \"$SUB\"\n",
        "\n",
        "echo\n",
        "echo \"Running MEGAHIT on subset...\"\n",
        "megahit \\\n",
        "  -r \"$SUB\" \\\n",
        "  -o \"$OUTDIR\" \\\n",
        "  --num-cpu-threads \"$THREADS\"\n",
        "\n",
        "echo\n",
        "echo \"MEGAHIT finished. Assembly:\"\n",
        "ls -lh \"$OUTDIR/final.contigs.fa\"\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "m3gb4ed93MAV"
      },
      "source": [
        "and now we run it on the carpenter bee"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "zXjbHHDs3Psu"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "# One Illumina dataset (single-end from R1)\n",
        "READS=\"/content/SRR24955310_1.fastq\"  # CHANGE PATHNAME IF NEEDED\n",
        "OUTDIR=\"/content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee/result_spades\"\n",
        "THREADS=8      # adjust if needed\n",
        "MEM_GB=12      # approximate RAM limit\n",
        "\n",
        "mkdir -p \"$OUTDIR\"\n",
        "\n",
        "spades.py \\\n",
        "  -s \"$READS\" \\\n",
        "  -o \"$OUTDIR\" \\\n",
        "  -t \"$THREADS\" \\\n",
        "  -m \"$MEM_GB\" \\\n",
        "  --careful\n",
        "\n",
        "echo \"SPAdes finished. Assembly:\"\n",
        "ls -lh \"$OUTDIR/contigs.fasta\"\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "ls -lh /content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee/result_spades/contigs.fasta"
      ],
      "metadata": {
        "id": "v0I8g4Uw3XpL"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "mv /content/CarpenterBee_fastq/SRR24955310_2.fastq /content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee"
      ],
      "metadata": {
        "id": "au0Z5_qcxO8s"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "rsync -ah --info=progress2 \\\n",
        "  /content/CarpenterBee_fastq/* \\\n",
        "  \"/content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee\"\n"
      ],
      "metadata": {
        "id": "qfMyv14FyvnA"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "rsync -ah --info=progress2 \\\n",
        "  /content/CarpenterBee_fastq/* \\\n",
        "  \"/content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee\"\n"
      ],
      "metadata": {
        "id": "ZIIzvRAmS9xB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "ls\n"
      ],
      "metadata": {
        "id": "07UyTX6R0GTC"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "DDs1f4z2U_r6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Megahit - Carpenter Bee\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "R1=\"/content/SRR24955310_1.sub500000.fq\"\n",
        "R2=\"/content/SRR24955310_2.sub500000.fq\"\n",
        "OUTDIR=\"/content/megahit_carpenterbee\"\n",
        "\n",
        "#echo \"Cleaning output dir ${OUTDIR}...\"\n",
        "#rm -rf \"$OUTDIR\"\n",
        "\n",
        "megahit \\\n",
        "  -1 \"$R1\" \\\n",
        "  -2 \"$R2\" \\\n",
        "  -o \"$OUTDIR\" \\\n",
        "  --min-count 2 \\\n",
        "  --k-min 21 \\\n",
        "  --k-max 81 \\\n",
        "  --k-step 10 \\\n",
        "  --mem-flag 2 \\\n",
        "  -t 8\n",
        "\n",
        "echo \"Assembly complete:\"\n",
        "ls -lh \"$OUTDIR/final.contigs.fa\""
      ],
      "metadata": {
        "id": "eGgRDellVTA2",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp -r /content/megahit_carpenterbee/ /content/drive/MyDrive/BioinformaticsProject/Reads/CarpenterBee/\n"
      ],
      "metadata": {
        "id": "ycNARvkMdOZZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Megahit - Honeybee\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "R1=\"/content/SRR36076821_1.sub500000.fq\"\n",
        "R2=\"/content/SRR36076821_2.sub500000.fq\"\n",
        "OUTDIR=\"/content/megahit_honeybee\"\n",
        "\n",
        "#echo \"Cleaning output dir ${OUTDIR}...\"\n",
        "#rm -rf \"$OUTDIR\"\n",
        "\n",
        "megahit \\\n",
        "  -1 \"$R1\" \\\n",
        "  -2 \"$R2\" \\\n",
        "  -o \"$OUTDIR\" \\\n",
        "  --min-count 2 \\\n",
        "  --k-min 21 \\\n",
        "  --k-max 81 \\\n",
        "  --k-step 10 \\\n",
        "  --mem-flag 2 \\\n",
        "  -t 8\n",
        "\n",
        "echo \"Assembly complete:\"\n",
        "ls -lh \"$OUTDIR/final.contigs.fa\"\n"
      ],
      "metadata": {
        "id": "-faPTv1mFg3y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Quast Evaluation"
      ],
      "metadata": {
        "id": "AiDPLbljD3-n"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Installation\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "micromamba install -y -c bioconda -c conda-forge quast"
      ],
      "metadata": {
        "collapsed": true,
        "id": "HG4eW4ct4re_"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Honeybee Reference Genome\n",
        "\n",
        "%%bash\n",
        "# Download honeybee reference genome (RefSeq)\n",
        "wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/254/395/GCF_003254395.2_Amel_HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.fna.gz\n",
        "gunzip GCF_003254395.2_Amel_HAv3.1_genomic.fna.gz\n"
      ],
      "metadata": {
        "collapsed": true,
        "id": "1J3vUuBDGowN"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Honeybee - Run Quast on Spades\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "REFERENCE=\"GCF_003254395.2_Amel_HAv3.1_genomic.fna\"\n",
        "ASSEMBLY=\"/content/spades_honeybee/contigs.fasta\"\n",
        "OUTDIR=\"/content/quast_honeybee_spades\"\n",
        "\n",
        "quast.py \"$ASSEMBLY\" \\\n",
        "  -r \"$REFERENCE\" \\\n",
        "  -o \"$OUTDIR\""
      ],
      "metadata": {
        "collapsed": true,
        "cellView": "form",
        "id": "3nDMqlewGojH"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "hUCObrP4D4Tc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Save to Drive\n",
        "!cp -r quast_honeybee_spades/ /content/drive/MyDrive/BioinformaticsProject/Reads/Honeybee/"
      ],
      "metadata": {
        "id": "jgL2U8JBBgV2",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp -r /content/drive/MyDrive/BioinformaticsProject/Reads/Honeybee/result_megahit /content/\n",
        "!mv /content/result_megahit /content/result_megahit_honeybee"
      ],
      "metadata": {
        "id": "sn9bo7I_DFGj"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Honeybee - Run Quast on Megahit\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "REFERENCE=\"GCF_003254395.2_Amel_HAv3.1_genomic.fna\"\n",
        "ASSEMBLY=\"/content/result_megahit_honeybee/final.contigs.fa\"\n",
        "OUTDIR=\"/content/quast_honeybee_megahit\"\n",
        "\n",
        "quast.py \"$ASSEMBLY\" \\\n",
        "  -r \"$REFERENCE\" \\\n",
        "  -o \"$OUTDIR\""
      ],
      "metadata": {
        "cellView": "form",
        "collapsed": true,
        "id": "cVe9qjD7C9P6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Save results\n",
        "\n",
        "!cp -r /content/quast_honeybee_megahit /content/drive/MyDrive/BioinformaticsProject/Reads/Honeybee"
      ],
      "metadata": {
        "id": "wLI-eTzuFNHc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "_gnpdTH1zWXQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Carpenter Bee Reference Genome - Install Datasets CLI\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "\n",
        "micromamba install -c conda-forge ncbi-datasets-cli\n"
      ],
      "metadata": {
        "collapsed": true,
        "id": "k7Oa8wLD0HVp"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Carpenter Bee Reference Genome - Pull from accession number\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "datasets download genome accession GCA_049004755.1 --include genome"
      ],
      "metadata": {
        "collapsed": true,
        "id": "8lwYn5b5ANxV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip ncbi_dataset.zip -d carpenterbee_reference_archive/\n"
      ],
      "metadata": {
        "id": "kQbXiGOIAl9c"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Carpenter Bee - Run Quast on Spades\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "REFERENCE=\"/content/carpenterbee_reference_archive/ncbi_dataset/data/GCA_049004755.1/GCA_049004755.1_ASM4900475v1_genomic.fna\"\n",
        "ASSEMBLY=\"/content/carpenterbee_megahit_contigs.fa\"\n",
        "OUTDIR=\"/content/quast_carpenterbee_spades\"\n",
        "\n",
        "quast.py \"$ASSEMBLY\" \\\n",
        "  -r \"$REFERENCE\" \\\n",
        "  -o \"$OUTDIR\""
      ],
      "metadata": {
        "collapsed": true,
        "id": "Onhsze7iKYmP"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Carpenter Bee - Run Quast on Megahit\n",
        "\n",
        "%%bash\n",
        "set -e\n",
        "\n",
        "# Activate micromamba env\n",
        "eval \"$(micromamba shell hook -s bash)\"\n",
        "micromamba activate beeasm\n",
        "\n",
        "REFERENCE=\"/content/carpenterbee_reference_archive/ncbi_dataset/data/GCA_049004755.1/GCA_049004755.1_ASM4900475v1_genomic.fna\"\n",
        "ASSEMBLY=\"/content/carpenterbee_megahit_contigs.fa\"\n",
        "OUTDIR=\"/content/quast_carpenterbee_megahit\"\n",
        "\n",
        "quast.py \"$ASSEMBLY\" \\\n",
        "  -r \"$REFERENCE\" \\\n",
        "  -o \"$OUTDIR\""
      ],
      "metadata": {
        "collapsed": true,
        "id": "QUGqxE6p4s52"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}